src/game/savegame.c

```diff
@@ -431,9 +432,15 @@ bool Savegame_UpdateDeathCounters(int32_t slot_num, GAME_INFO *game_info)
     return ret;
 }
 
+void Savegame_Init()
+{
+    m_SavegameInfo = Memory_Alloc(sizeof(*m_SavegameInfo) * g_Config.maximum_save_slots);
+    Option_PassportInitSavegameStrings();
+}
+
 void Savegame_Shutdown(void)
 {
-    for (int i = 0; i < MAX_SAVE_SLOTS; i++) {
+    for (int i = 0; i < g_Config.maximum_save_slots; i++) {
```

Comment:
Missing `Memory_FreePointer(&m_SavegameInfo)`

Reply:
And to make it work with the flipped init/shutdown order, we can check if `m_SavegameInfo` is not null before dereferencing it down below :)